examples/microprofile/openapi-basic/src/main/java/io/helidon/microprofile/examples/openapi/basic/GreetResource.java

```diff
@@ -0,0 +1,201 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.examples.openapi.basic;
+
+import java.util.Collections;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.json.Json;
+import javax.json.JsonBuilderFactory;
+import javax.json.JsonObject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.eclipse.microprofile.openapi.annotations.media.Content;
+import org.eclipse.microprofile.openapi.annotations.media.ExampleObject;
+import org.eclipse.microprofile.openapi.annotations.media.Schema;
+import org.eclipse.microprofile.openapi.annotations.parameters.RequestBody;
+import org.eclipse.microprofile.openapi.annotations.responses.APIResponse;
+
+/**
+ * A simple JAX-RS resource with OpenAPI annotations to greet you. Examples:
+ *
+ * Get default greeting message:
+ * curl -X GET http://localhost:8080/greet
+ *
+ * Get greeting message for Joe:
+ * curl -X GET http://localhost:8080/greet/Joe
+ *
+ * Change greeting
+ * curl -X PUT -H "Content-Type: application/json" -d '{"greeting" : "Howdy"}' http://localhost:8080/greet/greeting
+ *
+ * Get OpenAPI document for the endpoints
+ * curl -X GET http://localhost:8080/openapi
+ *
+ * Note that the output will include not only the annotated endpoints from this
+ * class but also an endpoint added by the {@link SimpleAPIModelReader}.
+ *
+ * The message is returned as a JSON object.
+ */
+@Path("/greet")
+@RequestScoped
+public class GreetResource {
+
+    private static final JsonBuilderFactory JSON = Json.createBuilderFactory(Collections.emptyMap());
+
+    /**
+     * The greeting message provider.
+     */
+    private final GreetingProvider greetingProvider;
+
+    /**
+     * Using constructor injection to get a configuration property.
+     * By default this gets the value from META-INF/microprofile-config
+     *
+     * @param greetingConfig the configured greeting message
+     */
+    @Inject
+    public GreetResource(GreetingProvider greetingConfig) {
+        this.greetingProvider = greetingConfig;
+    }
+
+    /**
+     * Return a worldly greeting message.
+     *
+     * @return {@link JsonObject}
+     */
+    @SuppressWarnings("checkstyle:designforextension")
```

Comment:
Can you please not put these annotations in the code? If you use Idea, you can simply suppress this warning for methods annotated with a specific annotation (probably in other IDEs there is a way for that as well).
We do not have a checkstyle warning that would fail on such a method, so it should not be in the sources.

Reply:
I'll remove those from this example. 

There are several other places in our examples -- notably the quickstarts. This example is based on the MP quickstart which is why these `@SuppressWarnings` annos were here.